### DISCUSSION TITLE
dart bindings for webgpu

### DISCUSSION THREAD

**Discussion by unicomp21** [Blimster/babylon_dart#17] [Ideas]
Would it be easy to code generate dart bindings for webgpu?  In addition to what we have for babylonjs/babylon_dart?  The reason for asking, we could then start using procedural generation technique for generating meshes and textures on the fly.  This opens up a huge possibility in that high levels of detail can be supported w/o needing to download and burn gigabytes/terabytes of bandwidth.  See Mojoworld in "Texturing and Modeling, a procedural approach, 3rd edition".

**Comment by unicomp21** [comment]
I think it's accurate to say we can now realize Mojoworld in the web browser?  At the same time, I'm a little sad Ken Musgrave isn't around to see his prophecy get fulfilled.

**Comment by Blimster** [comment]
All the magic behind babylon_dart is the generator. Currently, its feature set is focused on generating the bindings for BabylonJS. But I hope that the generator can be a standalone alternative for [js_facade_gen](https://github.com/dart-lang/js_facade_gen) in the future.

With that said and with [gpuweb/types](https://github.com/gpuweb/types) it should be possible to generate bindings for WebGPU.

PS: Currently, I'm working on a complete rewrite of the generator, because the first implementation has some limitations. But sadly, I have not much time for this. This results in slow progress.

PPS: I think, as soon the WebGPU is stable, the Dart team will add the API to the SDK.

**Comment by unicomp21** [comment]
Sadly, I don't think it's a priority, or ever will be.  Webgpu has been released in chrome for a while now.

https://github.com/dart-lang/sdk/issues/46710
